pretty_midi/pretty_midi.py

```diff
@@ -166,27 +168,34 @@ def _load_metadata(self, midi_data):
         """
 
         # Initialize empty lists for storing key signature changes, time
-        # signature changes, and lyrics
+        # signature changes, lyrics and a dictionary (start_time, lyrics index) for faster vocals matching
         self.key_signature_changes = []
         self.time_signature_changes = []
         self.lyrics = []
+        self.__lyrics_times = {}
 
-        for event in midi_data.tracks[0]:
-            if event.type == 'key_signature':
-                key_obj = KeySignature(
-                    key_name_to_key_number(event.key),
-                    self.__tick_to_time[event.time])
-                self.key_signature_changes.append(key_obj)
-
-            elif event.type == 'time_signature':
-                ts_obj = TimeSignature(event.numerator,
-                                       event.denominator,
-                                       self.__tick_to_time[event.time])
-                self.time_signature_changes.append(ts_obj)
-
-            elif event.type == 'lyrics':
-                self.lyrics.append(Lyric(
-                    event.text, self.__tick_to_time[event.time]))
+        # We search for metadata events on all tracks
```

Comment:
My understanding is that if key signature or time signature events appearing on a track other than 0, the MIDI file should be treated as corrupt. Do you have any reference to the contrary?

Reply:
In some files text events appear on other channels (e.g. 69df087321438541ae7c8f064b109917.mid).